### ISSUE TITLE
Broken with Guard 2.8+

### ISSUE THREAD

**Issue by colszowka** [vizjerai/guard-spin#17]
Apparently Guard 2.8 introduced a bunch of API changes.

On 2.9, I get this error:

```
10:34:53 - ERROR - Could not load 'guard/spin' or'    ' find class Guard::Spin
10:34:53 - ERROR - Error is: cannot load such file -- guard/guard
10:34:53 - ERROR - ...ruby/2.1.5/gems/guard-spin-0.3.0/lib/guard/spin.rb:2:in `require'
```

On 2.8, there is this massive deprecation warning:

```
    ******** BIG DEPRECATION WARNING !! ********

    Hi, Guard here.

    You're including lib/guard/guard.rb ...

    ... which contains code deprecated over a year ago!


    This file will likely be removed in the next version, so make sure you're
    not requiring it anywhere to ensure safe gem upgrades.

    If this message is annoying and you can't quickly fix the issue (see below),
    you have 2 options:

      1) Simply set the env variable "GUARD_GEM_SILENCE_DEPRECATIONS" to "1" to
      skip this message

      2) Freeze the gem to a previous version (not recommended because upgrades
      are cool and you might forget to unfreeze later!).

      E.g. in your Gemfile:

        if Time.now > Time.new(2014,11,10)
          gem 'guard', '~> 2.8'
        else
          # Freeze until 2014-11-10 - in case we forget to change back ;)
          gem 'guard', '= 2.7.3'
        end

    If you don't know which gem or plugin is requiring this file, here's a
    backtrace:

    .../.gem/ruby/2.1.5/gems/guard-2.8.2/lib/guard/guard.rb:45:in `<module:Guard>'
	 >>     .../.gem/ruby/2.1.5/gems/guard-2.8.2/lib/guard/guard.rb:3:in `<top (required)>'
	 >>     .../.gem/ruby/2.1.5/gems/guard-spin-0.3.0/lib/guard/spin.rb:2:in `require'
	 >>     .../.gem/ruby/2.1.5/gems/guard-spin-0.3.0/lib/guard/spin.rb:2:in `<top (required)>'
	 >>     .../.gem/ruby/2.1.5/gems/guard-2.8.2/lib/guard/plugin_util.rb:123:in `require'"

    Here's how to quickly upgrade/fix this (given you are a maintainer of the
    offending plugin or you want to prepare a pull request yourself):

      https://github.com/guard/guard/wiki/Upgrading-to-Guard-2.0#changes-in-guardguard

    Have fun!

    ******** END OF DEPRECATION MESSAGE ********
```

**Comment by colszowka** [comment]
This would be resolved by https://github.com/vizjerai/guard-spin/pull/18, I have tried pulling in the gem from that branch and it works flawlessly again.

**Comment by jgeiger** [comment]
Merged #18
